tests/components/plugwise/test_config_flow.py

```diff
@@ -386,3 +389,104 @@ async def test_zeroconf_abort_anna_with_adam(hass: HomeAssistant) -> None:
     flows_in_progress = hass.config_entries.flow.async_progress()
     assert len(flows_in_progress) == 1
     assert flows_in_progress[0]["context"]["product"] == "smile_open_therm"
+
+
+async def test_form_invalid_setup(hass: HomeAssistant, mock_smile: MagicMock) -> None:
+    """Test we handle invalid setup."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+    )
+
+    mock_smile.connect.side_effect = InvalidSetupError
+    mock_smile.gateway_id = "0a636a4fc1704ab4a24e4f7e37fb187a"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_HOST: TEST_HOST, CONF_PASSWORD: TEST_PASSWORD},
+    )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_setup"}
```

Comment:
Hmm how would a user recover from an invalid setup?

Reply:
Ok, thanks. I now realize that the added tests correspond to changes I made to the config_flow code (a long time ago) that also could be added to the Core Plugwise code.
I will remove these test-cases from this PR and add them in a new, more appropriate PR.

Reply:
Test-case removed.

Reply:
To answer you're first question, that functionality will be added in the new PR.